### ISSUE TITLE
Downloads are not working / Program doesn't do anything

### ISSUE THREAD

**Issue by Csfop7** [MalloyDelacroix/DownloaderForReddit#153]
As said in title, after launching the program and starting any kind of download, it runs for a couple seconds (5-8) and then just stops with this output:
![DownloaderForReddit_cwwlObjL8Z](https://user-images.githubusercontent.com/72454464/95214281-0a393980-07f0-11eb-9b59-15480ca956c5.png)

This time I tested it with the subreddit "nosleep" and no modified settings, just after extracting and launching the program "for the first time" after clearing the Roaming folder. I've tried it with different subreddits, users, download settings, directories imgur settings as well as with or without ffmpeg. Always the same result.

**Environment Information**
 - Operating system: Win10 64 Home
 - Software version: v3.2.1-beta
 - Are you running a compiled version or from source?: Tried both, same results.

**To Reproduce** (optional)
Steps to reproduce the behavior (seems to only apply to my case though):
1. Launch the program
2. Attempt to download anything
3. Program runs for a couple seconds and then "finishes" without downloading anything
4. No error message

**For all but the most trivial of issues, please attach the latest log file.**

> {"levelname": "INFO", "name": "DownloaderForReddit.DownloaderForReddit.database.migration", "filename": "migration.py", "module": "migration", "funcName": "check_migration", "lineno": 43, "message": "Migration not performed: no version information found in database.  Database entryfor version v3.2.1-beta has been created.", "asctime": "10/06/2020 04:18:43 PM"}
{"levelname": "INFO", "name": "DownloaderForReddit.DownloaderForReddit.gui.downloader_for_reddit_gui", "filename": "downloader_for_reddit_gui.py", "module": "downloader_for_reddit_gui", "funcName": "run", "lineno": 539, "message": "Download thread started", "asctime": "10/06/2020 04:19:16 PM"}
{"levelname": "WARNING", "name": "DownloaderForReddit.utils.video_merger", "filename": "video_merger.py", "module": "video_merger", "funcName": "merge_videos", "lineno": 89, "message": "Ffmpeg is not installed: unable to merge video and audio files", "asctime": "10/06/2020 04:19:22 PM", "videos_to_merge": 0}
{"levelname": "INFO", "name": "DownloaderForReddit.DownloaderForReddit.core.download_runner", "filename": "download_runner.py", "module": "download_runner", "funcName": "finish_messages", "lineno": 479, "message": "Download complete", "asctime": "10/06/2020 04:19:22 PM", "download_time": "5 secs", "significant_user_count": 0, "significant_subreddit_count": 1, "total_user_count": 1, "total_subreddit_count": 25, "post_extraction_count": 25, "comment_extraction_count": 0, "download_count": 0}

------------------------------------------------------------------------------------------------------------

Thanks for reading.

**Comment by MalloyDelacroix** [comment]
Are the posts you are trying to download all self posts?  The default settings are to not download self post text.  The text is only extracted and stored in the database.  I can see by the screen shot and log you've posted that there were 25 posts extracted, so I believe this is the problem.  At least with the nosleep subreddit.  

This can be adjusted in the settings menu for the user/subreddit or can be applied to the whole list.

**Comment by Csfop7** [comment]
Okay, I tried experimenting some more and found this: first time downloads after adding a subreddit work fine (the issue with nosleep really was self posts). However, any subsequent downloads don't go through, because a date limit that's actually ahead of present time gets automatically enabled every time I modify subreddit settings.
![DownloaderForReddit_Orto8da0Cc](https://user-images.githubusercontent.com/72454464/95411733-8b471c80-0927-11eb-9256-fdc554a6917f.png)

Turning it off doesn't work, as it gets turned on automatically upon closing the settings window. I can work around this if I manually set a date limit.

That makes downloads work... mostly. The downloader likes to just get stuck on downloads though.
